jtransc-core/src/com/jtransc/plugin/reflection/MetaReflectionJTranscPlugin.kt

```diff
@@ -487,17 +487,18 @@ class MetaReflectionJTranscPlugin : JTranscPlugin() {
 
 			data class MemberInfoWithRef(val ref: Any, val mi: MemberInfo)
 
-			fun genMemberMethods(list: List<Pair<AstClass, List<MemberInfoWithRef>>>, clazz: AstClass, method: AstMethod): List<AstMethod> {
+			fun genMemberMethods(list: List<Pair<AstClass, List<MemberInfoWithRef>>>, classes: List<AstClass>, methodName: String): List<AstMethod> {
+
 				val additionalMethods: MutableList<AstMethod> = mutableListOf()
 				var methodIndex: Int = 0
 				while (methodIndex * CASES_PER_SWITCH < list.size) {
 					val mI: Int = methodIndex
 					val sI: Int = methodIndex * CASES_PER_SWITCH
-					val parentClass: AstClass = clazz
-					val parentMethod: AstMethod = method
+					val parentClass: AstClass = classes[methodIndex % classes.size]
+					val parentMethod: AstMethod? = parentClass.getMethodWithoutOverrides(methodName)
```

Comment:
Have you tested this? Maybe we should add a test to big test with CASES_PER_SWITCH * 3 or things like that and tons of methods and fields and check reflection. The problem is that tests may pass with a few classes or small classes and production code with tons of classes fail, or not work as expected with some edge cases